ads/google/a4a/traffic-experiments.js

```diff
@@ -121,7 +121,16 @@ export function googleAdsIsA4AEnabled(win, element, experimentName,
  */
 function maybeSetExperimentFromUrl(win, experimentName,
     controlBranchId, treatmentBranchId, manualId) {
-  const expParam = parseQueryString(win.location.search)['exp'];
+  let queryString = win.location.search;
+  // If we're inside the AMP viewer, the hash fragment will have been moved
+  // to the originalHash field.
+  const hash = win.location.originalHash || win.location.hash;
```

Comment:
Yes. If `exp` is the only thing you want, `viewer.getParam('exp')` is a better choice. Right now `originalHash` is only used to defer `mode`. Viewer param can be passed to runtime in multiple places, see this [code](https://github.com/ampproject/amphtml/blob/master/src/service/viewer-impl.js#L177). 

Please see previous [discussion](https://github.com/ampproject/amphtml/pull/4079#discussion_r70905166)